shuup/simple_supplier/utils.py

```diff
@@ -147,7 +147,13 @@ def get_stock_adjustment_div(request, supplier, product):
         "product": product,
         "supplier": supplier,
         "delta_step": pow(0.1, product.sales_unit.decimals) if product.sales_unit.decimals else 0,
-        "adjustment_form": StockAdjustmentForm(initial={"purchase_price": purchase_price, "delta": None}),
+        "adjustment_form": StockAdjustmentForm(
+            initial={
+                "purchase_price": purchase_price,
+                "delta": None,
+                "decimal_places": product.sales_unit.decimals,
```

Comment:
Passing decimals instead of sales_unit as we need to recover it from the POST data also in a hidden field.